cinema/views.py

```diff
@@ -1,45 +1,121 @@
-            serializer.save()
-            return Response(serializer.data, status=status.HTTP_201_CREATED)
 
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+class GenreList(APIView):
+    def get(self, request):
+        genres = Genre.objects.all()
+        serializers = GenreSerializer(genres, many=True)
+
+        return Response(serializers.data, status=status.HTTP_200_OK)
+
+    def post(self, request: {data}) -> Response:
+        serializers = GenreSerializer(data=request.data)
+        if serializers.is_valid():
+            serializers.save()
+            return Response(serializers.data, status=status.HTTP_201_CREATED)
 
+        return Response(serializers.errors, status=status.HTTP_400_BAD_REQUEST)
 
-@api_view(["GET", "PUT", "DELETE"])
-def movie_detail(request, pk):
-    movie = get_object_or_404(Movie, pk=pk)
 
-    if request.method == "GET":
-        serializer = MovieSerializer(movie)
-        return Response(serializer.data, status=status.HTTP_200_OK)
+class GenreDetail(APIView):
+    def get_object(self, pk: int):
+        try:
+            return Genre.objects.get(pk=pk)
+        except Genre.DoesNotExist:
+            raise Http404
 
-    if request.method == "PUT":
-        serializer = MovieSerializer(movie, data=request.data)
+    def get(self, request, pk: int):
+        genre = self.get_object(pk=pk)
+        serializers = GenreSerializer(genre)
+        return Response(serializers.data, status=status.HTTP_200_OK)
+
+    def put(self, request, pk: int):
+        genre = self.get_object(pk=pk)
+        serializers = GenreSerializer(genre, data=request.data)
+        if serializers.is_valid():
+            serializers.save()
+            return Response(serializers.data, status=status.HTTP_200_OK)
+
+        return Response(serializers.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def patch(self, request, pk):
+        genre = self.get_object(pk)
+        serializer = GenreSerializer(genre, data=request.data, partial=True)
         if serializer.is_valid():
             serializer.save()
             return Response(serializer.data, status=status.HTTP_200_OK)
 
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
-    if request.method == "DELETE":
-        movie.delete()
+    def delete(self, request, pk: int):
+        genre = self.get_object(pk=pk)
+        genre.delete()
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class ActorList(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    generics.GenericAPIView
+):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+    def get(self, request, *args, **kwargs):
+        return self.list(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        return self.create(request, *args, **kwargs)
+
+
+class ActorDetail(
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
+    mixins.DestroyModelMixin,
```

Comment:
You can use generics.RetrieveUpdateDestroyAPIView
Also no need to change get, put, patch, delete